nntrainer/src/optimizer.cpp

```diff
@@ -136,20 +127,21 @@ void Optimizer::apply_gradients(std::shared_ptr<UpdatableParam> params,
       x.add_i(x_grad, -ll);
       break;
     case OptType::adam: {
-      std::function<float(float)> sqrtEps = [&](float f) {
-        return sqrtFloat(f) + this->popt.epsilon;
-      };
 
       Tensor wm = weight_mv[idx].first;
       Tensor wv = weight_mv[idx].second;
 
       wm.multiply_i(popt.beta1);
-      wm.add_i(x_grad, 1 - popt.beta1);
+      wm.add_i(x_grad, 1.0f - popt.beta1);
 
       wv.multiply_i(popt.beta2);
-      wv.add_i(x_grad.multiply(x_grad), 1 - popt.beta2);
+      wv.add_i(x_grad.multiply(x_grad), 1.0f - popt.beta2);
+
+      Tensor denom = wv.apply(sqrtFloat)
+                       .divide(sqrtFloat(biasCorrection2))
+                       .add(popt.epsilon);
+      x.add_i(wm.divide(denom), -ll / biasCorrection1);
```

Comment:
This looks the same as yesterday.
I dont understand what was the bug.
Can you please share?

Reply:
I thought that it was wrong. Because there was diff between tf and ours. ( Not many, but two values as below ) That's what I implemented differently. But I guess that was right one. 
```
[ RUN      ] nntrainer_FullyConnectedLayer_TFmatch.forwarding_backwarding_00_p
../test/unittest/unittest_nntrainer_layers.cpp:89: Failure
The difference between result[i] and golden_ptr[i] is 0.25586135685443878, which exceeds 10e-5, where
result[i] evaluates to -0.1319877952337265,
golden_ptr[i] evaluates to -0.38784915208816528, and
10e-5 evaluates to 0.0001.
../test/unittest/unittest_nntrainer_layers.cpp:89: Failure
The difference between result[i] and golden_ptr[i] is 0.25586199760437012, which exceeds 10e-5, where
result[i] evaluates to -0.057956874370574951,
golden_ptr[i] evaluates to -0.31381887197494507, and
10e-5 evaluates to 0.0001.
[  FAILED  ] nntrainer_FullyConnectedLayer_TFmatch.forwarding_backwarding_00_p (5 ms)
```

Reply:
Thats quite strange. It looks like some indexing issue.